### ISSUE TITLE
How does `solve_max_rooms` work? Is `compose_indoors.solve_max_rooms` deprecated?

### ISSUE THREAD

**Issue by GCChen97** [princeton-vl/infinigen#296]
I am trying to generate one-room scene but confused by the parameter `solve_max_rooms` of `compose_indoors` (in command line) and `restrict_solving` (in `singleroom.gin`).

## Steps to Reproduce
1. I used `singleroom.gin` in the first two commands in the following and `compose_indoors.solve_max_rooms=1` in the third one, but only the second one generate a one-room scene.
```
# LivingRoom, single room only, first person view (~11min runtime, CPU only)
python -m infinigen_examples.generate_indoors --seed 0 --task coarse --output_folder outputs/indoors/coarse \
-g fast_solve.gin singleroom.gin \
-p compose_indoors.terrain_enabled=False \
restrict_solving.restrict_parent_rooms=\[\"LivingRoom\"\]

# Floor layout, overhead view, no objects (~34 second runtime, CPU only):
python -m infinigen_examples.generate_indoors --seed 0 --task coarse --output_folder outputs/indoors/coarse \
-g no_objects.gin overhead.gin singleroom.gin \
-p compose_indoors.terrain_enabled=False

# Floor layout, overhead view, no objects (~34 second runtime, CPU only):
python -m infinigen_examples.generate_indoors --seed 0 --task coarse --output_folder outputs/indoors/coarse \
-g no_objects.gin overhead.gin \
-p compose_indoors.terrain_enabled=False \
compose_indoors.solve_max_rooms=1
```

2. I removed `compose_indoors.solve_max_rooms=1` in the second command in the follows, but both outputted one-room scenes.
```
# Single random room with objects, overhead view (~11min. runtime CPU only):
python -m infinigen_examples.generate_indoors --seed 0 --task coarse --output_folder outputs/indoors/coarse \
-g fast_solve.gin overhead.gin singleroom.gin \
-p compose_indoors.terrain_enabled=False \
compose_indoors.overhead_cam_enabled=True \
compose_indoors.solve_max_rooms=1 \
compose_indoors.invisible_room_ceilings_enabled=True \
compose_indoors.restrict_single_supported_roomtype=True

python -m infinigen_examples.generate_indoors --seed 0 --task coarse --output_folder outputs/indoors/coarse \
-g fast_solve.gin overhead.gin singleroom.gin \
-p compose_indoors.terrain_enabled=False \
compose_indoors.overhead_cam_enabled=True \
compose_indoors.invisible_room_ceilings_enabled=True \
compose_indoors.restrict_single_supported_roomtype=True
```

Based on the outputs by the above 5 commands, `compose_indoors.solve_max_rooms` seems to be deprecated. And I could only find `solve_max_rooms` in the codes related to `restrict_restrict_solving` but not `compose_indoors`.

So, how to create a one-room `LivingRoom` or other types of rooms without any extra empty rooms?

### What version of the code were you using? 
Tell us the commit & commit hash from `git log`
```
commit 4a4411e92611d059f2476f4384141de3529d1f9f (HEAD -> main, origin/main, origin/HEAD)
Merge: 8d079368 792d170b
Author: Alexander Raistrick <araistrick@princeton.edu>
Date:   Sun Aug 4 16:46:28 2024 -0400

    Merge branch 'develop'

commit 792d170b246cbceee86acecfef39889b677ed1bd
Merge: 0712df61 54e72d4e
Author: Zeyu Ma <31351547+mazeyu@users.noreply.github.com>
Date:   Fri Aug 2 00:10:34 2024 -0400

    Merge pull request #236 from princeton-vl/hotfix_setattr
    
    hotfix nested setattr (previously through exec)

commit 54e72d4e6ca4aba24ceeb541b24d2a5ec00e0ca8
Author: Zeyu Ma <zeyum@princeton.edu>
Date:   Thu Aug 1 23:23:30 2024 -0400

    hotfix nested setattr (previously through exec)

commit 8d079368c3546bc4b4002dd4a91f649c210583eb (tag: v1.6.0)
Merge: c537e245 0712df61
Author: pvl-bot <bot@infinigen.org>
Date:   Thu Aug 1 18:53:56 2024 -0400

    Merge v1.6.0 from branch 'develop'

commit 0712df61c260fba3e664867c0ef2ca6ff9d6b8b6
Merge: 8c4e30d0 ef328ae4
Author: Alex Raistrick <araistrick@princeton.edu>
Date:   Thu Aug 1 18:41:42 2024 -0400

    Merge pull request #225 from princeton-vl/mesher_version_check
    
    Mesher version check

commit 8c4e30d0dfeac3121861a854ffdbb0a9861aa4d6
Merge: 787feeb1 84880b8f
Author: Alex Raistrick <araistrick@princeton.edu>
Date:   Thu Aug 1 18:36:55 2024 -0400

    Merge pull request #230 from princeton-vl/f_more_format_rules
    
    Format rules for copyright requirements & import
```

### Platform
- OS & OS Version: Ubuntu 22.04

**Comment by araistrick** [comment]
`compose_indoors.solve_max_rooms` is indeed deprecated / typod, it should always be `restrict_solving.solve_max_rooms`. I will fix the mentions of the former.

This setting will affect how many rooms receive furniture, as furniture placement is slow and is proportional to the number of rooms solved. It doesnt affect how many room meshes are generated, so in the current code there will always be some empty rooms surrounding the single room that recieved furniture. 

The empty room meshes dont add much to the cost, and without them you would get sunlight shining into the scene from open doorways etc, so I recommend just leaving them. If you specifically dont want the empty room meshes for the rest of the house, you can loop through and delete the objects using your own python code at the end of `compose_indoors`.

**Comment by araistrick** [comment]
I believe the typo mentioned in this issue is resolved by v1.8.1. Please re-open if you have further questions.
